src/vcpkg/commands.help.cpp

```diff
@@ -116,75 +69,46 @@ namespace
-        {"contact", command_topic_fn<CommandContactMetadata>},
-        {"deactivate", command_topic_fn<CommandDeactivateMetadata>},
-        {"depend-info", command_topic_fn<CommandDependinfoMetadata>},
-        {"x-download", command_topic_fn<CommandDownloadMetadata>},
-        {"edit", command_topic_fn<CommandEditMetadata>},
-        {"env", command_topic_fn<CommandEnvMetadata>},
-        {"export", command_topic_fn<CommandExportMetadata>},
-        {"fetch", command_topic_fn<CommandFetchMetadata>},
-        {"find", command_topic_fn<CommandFindMetadata>},
-        {"format-manifest", command_topic_fn<CommandFormatManifestMetadata>},
-        {"hash", command_topic_fn<CommandHashMetadata>},
-        {"help", command_topic_fn<CommandHelpMetadata>},
-        {"x-init-registry", command_topic_fn<CommandInitRegistryMetadata>},
-        {"install", command_topic_fn<CommandInstallMetadata>},
-        {"integrate", integrate_topic_fn},
-        {"list", command_topic_fn<CommandListMetadata>},
-        {"new", command_topic_fn<CommandNewMetadata>},
-        {"owns", command_topic_fn<CommandOwnsMetadata>},
-        {"x-package-info", command_topic_fn<CommandPackageInfoMetadata>},
-        {"portsdiff", command_topic_fn<CommandPortsdiffMetadata>},
-        {"x-regenerate", command_topic_fn<CommandRegenerateMetadata>},
-        {"remove", command_topic_fn<CommandRemoveMetadata>},
-        {"search", command_topic_fn<CommandSearchMetadata>},
-        {"x-set-installed", command_topic_fn<CommandSetInstalledMetadata>},
         {"topics", help_topics},
         {"triplet", [](const VcpkgPaths& paths) { help_topic_valid_triplet(paths.get_triplet_db()); }},
-        {"x-update-baseline", command_topic_fn<CommandUpdateBaselineMetadata>},
-        {"x-update-registry", command_topic_fn<CommandUpdateRegistryMetadata>},
-        {"update", command_topic_fn<CommandUpdateMetadata>},
-        {"upgrade", command_topic_fn<CommandUpgradeMetadata>},
-        {"use", command_topic_fn<CommandUseMetadata>},
-        {"version", command_topic_fn<CommandVersionMetadata>},
         {"versioning", help_topic_versioning},
-        {"x-vs-instances", command_topic_fn<CommandVsInstancesMetadata>},
     };
 
     void help_topics(const VcpkgPaths&)
     {
-        auto msg = msg::format(msgAvailableHelpTopics);
+        std::vector<std::string> all_topic_names;
         for (auto&& topic : topics)
         {
-            msg.append_raw(fmt::format("\n  {}", topic.name));
+            all_topic_names.push_back(topic.name.to_string());
+        }
+
+        for (auto&& command_metadata : get_all_commands_metadata())
+        {
+            all_topic_names.push_back(command_metadata->name.to_string());
+        }
+
+        Util::sort(all_topic_names);
+
+        auto msg = msg::format(msgAvailableHelpTopics);
+        for (auto&& topic_name : all_topic_names)
+        {
+            msg.append_raw(fmt::format("\n  {}", topic_name));
```

Comment:
Should this use `LocalizedString::append_indent()`?

Reply:
```suggestion
            msg.append_raw("\n").append_indent().append_raw(topic_name);
```

Reply:
1. Yes, I'll change it.
2. GitHub diff fail, I didn't meaningfully change this line from line 177 in the old code.